app/src/main/java/org/p2p/wallet/striga/signup/ui/StrigaSignUpFirstStepPresenter.kt

```diff
@@ -142,17 +150,37 @@ class StrigaSignUpFirstStepPresenter(
                 selectedPhoneNumber = selectedPhoneNumber
             )
         } else {
-            // find default country code for phone
             try {
-                phoneCountryCode = countryRepository.detectCountryOrDefault().also { countryCode ->
-                    val selectedPhoneNumber = signupData[StrigaSignupDataType.PHONE_NUMBER]?.value
-
-                    onPhoneCountryCodeChanged(countryCode)
-                    view?.setupPhoneCountryCodePicker(
-                        selectedCountryCode = countryCode,
-                        selectedPhoneNumber = selectedPhoneNumber
-                    )
+                // 1. getting number from db
+                var selectedPhoneNumber = signupData[StrigaSignupDataType.PHONE_NUMBER]?.value
+
+                // 2. getting code from db
+                var phoneCode = getCachedPhoneCodeWithoutPlus()?.let {
+                    countryRepository.findCountryCodeByPhoneCode(it)
+                }
+
+                // 3. if code is null, attempting to get phone code from metadata
+                // if phone number is not set, then we will use metadata phone number
+                val metadata = metadataInteractor.currentMetadata
```

Comment:
This doesn't look like presentation logic. Maybe we can extract this thing to a class and test it?